factiondraft.ttslua

```diff
@@ -0,0 +1,807 @@
+
+         if(reachTmp >= minReach) then
+            return true
+         end
+
+         tTemp[i] = nil
+         if( validateSetup(tTemp, reachTmp, playersLeft-1) ) then
+            return true
+         else
+            possible[i] = nil
+         end
+      end
+   end
+end
+
+function selectFaction(player, value, elem)
+   
+   local nextID = ""
+   local nextName = ""
+   local currentOrderIndex = -1
+   for i, c in ipairs(selectionOrder) do
+      if selections[c.id] == nil then
+         nextID = c.id
+         nextName = c.name
+         currentOrderIndex = i
+         break
+      end
+   end
+
+
+   if player.steam_id ~= nextID and #(tostring(nextID)) > 1 then
+      broadcastToColor("Not your turn yet", player.color)
+      return
+   end
+
+   local faction = factions[tonumber(value)]
+   if testPlayers then
+      selections[nextID] = faction
+   else
+      selections[player.steam_id] = faction
+   end
+   currentReach = currentReach + faction.reachVal
+
+   self.UI.setAttribute("reach-counter", "text", "Reach: " .. currentReach .. "/" .. reachMins[#selectionOrder])
+   self.UI.setAttribute("reach-counter", "color", "#dcdccc")
+
+   self.UI.hide("btn-faction-"..value)
+   local bgColor = self.UI.getAttribute("btn-faction-"..value, "color")
+   local fgColor = self.UI.getAttribute("btn-faction-"..value, "textColor")
+   local text = self.UI.getAttribute("btn-faction-"..value, "text")
+   self.UI.setAttribute("btn-faction-"..value, "color", fgColor)
+   self.UI.setAttribute("btn-faction-"..value, "text", text .. "\n" .. nextName .. " (" .. currentOrderIndex .. ")")
+   self.UI.setAttribute("btn-faction-"..value, "textColor", bgColor)
+   self.UI.setAttribute("btn-faction-"..value, "interactable", false)
+   self.UI.show("btn-faction-"..value)
+   
+   table.insert(turnOrder, 1, faction.color)
+
+   if value == "4" then
+      self.UI.show("btn-faction-8")
+   end
+
+   local playmat = getNextPlaymat()
+   
+   Wait.frames(
+      function()
+         if playmat == nil then
+            return
+         end
+         playmat.call("spawnRemote", {
+                         player = Player[faction.color],
+                         faction = faction.setupName,
+         })
+         Wait.frames(
+            function()
+               if value == "8" then
+                  playmat.call("setColorRemote", faction.color)
+               end
+            end,
+            3
+         )
+         player.changeColor(faction.color)
+      end,
+      30
+   )
+   
+   if selectionOrder[currentOrderIndex+1] ~= nil then
+      self.UI.setAttribute("turn-display", "text", selectionOrder[currentOrderIndex+1].name .. "'s Pick")
+      hideUnavailable()
+   else
+      self.UI.setAttribute("turn-display", "text", "Draft Finished! GLHF!")
+      Wait.frames(endDraft, 7)
+   end
+
+   
+end
+
+function getNextPlaymat()
+   for i, playmat in ipairs(playmats) do
+      if playmat ~= "" then
+         playmats[i] = ""
```

Comment:
```suggestion
      if playmat then
         playmats[i] = nil
```

Reply:
This does not work because if it is nil then it will break the ipairs looping needed